### ISSUE TITLE
typescript definitions for Ref values incorrect

### ISSUE THREAD

**Issue by bardkalbakk** [vueuse/vueuse#4255]
### Describe the bug

From version 11.1.0 typescript definitions for Ref values are wrong.  Type parameter is repeated twice for all Ref values.  This is consistent across the definition files index.d.mts, index.d.cts and index.ts.

Example from index.d.mts in version 11.1.0:
```
declare function useWindowSize(options?: UseWindowSizeOptions): {
    width: vue.Ref<number, number>;
    height: vue.Ref<number, number>;
};
```

from index.d.mts in version 11.0.3:
```
declare function useWindowSize(options?: UseWindowSizeOptions): {
    width: vue_demi.Ref<number>;
    height: vue_demi.Ref<number>;
};
```

### Reproduction

npm i @vueuse/core

### System Info

```Shell
System:
    OS: Windows 11 10.0.22621
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i9-12900HK
    Memory: 8.09 GB / 31.68 GB
  Binaries:
    Node: 20.11.1 - C:\Program Files\nodejs\node.EXE
    npm: 10.2.4 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (127.0.2651.86)
  npmPackages:
    @vueuse/core: ^11.1.0 => 11.1.0 
    vue: ^3.4.38 => 3.4.38
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by faga295** [comment]
Maybe related pr: https://github.com/vuejs/core/pull/11442

**Comment by surya76657** [comment]
any updates regarding this facing similar issue of typescript

**Comment by bardkalbakk** [comment]
Yes this issue is solved upgrading to vue 3.5 which has this new type definition.
